src/unsatcores/UnsatCoreBuilder.cc

```diff
@@ -53,11 +101,94 @@ void UnsatCoreBuilder::computeTerms() {
 
     Partitions partitions;
     for (CRef cref : clauses) {
-        auto const & partition = partitionManagerPtr->getClauseClassMask(cref);
+        auto const & partition = partitionManager.getClauseClassMask(cref);
         orbit(partitions, partitions, partition);
     }
 
-    terms = partitionManagerPtr->getPartitions(partitions);
+    terms = partitionManager.getPartitions(partitions);
+}
+
+void UnsatCoreBuilder::computeNamedTerms() {
+    assert(terms.size() > 0);
+    namedTerms.clear();
+    namedTermsIdxs.clear();
+
+    if (termNames.empty()) return;
+
+    size_t const termsSize = terms.size();
+    for (size_t idx = 0; idx < termsSize; ++idx) {
+        PTRef term = terms[idx];
+        if (not termNames.contains(term)) { continue; }
+        namedTerms.push(term);
+        namedTermsIdxs.push_back(idx);
+    }
+}
+
+void UnsatCoreBuilder::minimize() {
+    minimizeInit();
+    minimizeAlg();
+    minimizeFinish();
+}
+
+void UnsatCoreBuilder::minimizeInit() {
+    assert(terms.size() >= namedTerms.size());
+    assert(size_t(namedTerms.size()) == namedTermsIdxs.size());
+
+    setSmtSolverConfig();
+    setSmtSolver();
+}
+
+void UnsatCoreBuilder::minimizeAlgNaive() {
```

Comment:
To me it seems like it would be better if the new config and new solver were local to this method. Then, there would be no need for the `set` methods.

Reply:
But then it would not be usable in functions other than `minimizeAlgNaive`, and I plan to add more functions and select one of them based on the configuration.

Reply:
Then each method can have its own local solver.
The solver is not supposed to be shared between different methods, no?

Reply:
My concerns are readability and simplicity.

Reply:
Another thought: minimization stuff can be enclosed in a nested class (e.g. `Minimize`) where common functionalities among various strategies can be shared using inheritance, and member variables within the class can be "local" in a sense - within the scope of minimization. Also `minimizeInit` may be replaced by constructor, `minimizeFinish` by destructor, etc. It would still not use the variables as local variables within a function, but the scope of the variables would still be limited.

Reply:
How about we merge this now and you can rethink the design when you want to add another minimization strategy?

Reply:
I got a similar idea. I will make it local for now and we will see what to do next.